service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -111,6 +112,29 @@ public boolean deleteResource(UUID workspaceId, UUID resourceId) {
     return deleted;
   }
 
+  @WriteTransaction
+  public boolean deleteResourceForResourceType(
```

Comment:
I like this approach, but it's now a little awkward that only the "delete BQ table reference" endpoint enforces types. I think it's out of scope for this PR, but could you file a ticket to move the other delete endpoints to use this method?

Reply:
I added TODO at the callsite. I will file a ticket.